lib/web3/contract.js

```diff
@@ -166,40 +74,37 @@ var contract = function (abi) {
     return Contract.bind(null, abi);
 };
 
-function Contract(abi, options) {
+var Contract = function (abi, options) {
 
-    // workaround for invalid assumption that method.name is the full anonymous prototype of the method.
-    // it's not. it's just the name. the rest of the code assumes it's actually the anonymous
-    // prototype, so we make it so as a workaround.
-    // TODO: we may not want to modify input params, maybe use copy instead?
-    abi.forEach(function (method) {
-        if (method.name.indexOf('(') === -1) {
-            var displayName = method.name;
-            var typeName = method.inputs.map(function(i){return i.type; }).join();
-            method.name = displayName + '(' + typeName + ')';
-        }
-    });
-
-    var address = '';
+    this.address = '';
+    this._isTransaction = null;
+    this._options = {};
     if (utils.isAddress(options)) {
-        address = options;
+        this.address = options;
     } else { // is an object!
         // TODO, parse the rest of the args
         options = options || {};
         var args = Array.prototype.slice.call(arguments, 2);
         var bytes = solAbi.formatConstructorParams(abi, args);
         options.data += bytes;
-        address = web3.eth.sendTransaction(options);
+        this.address = web3.eth.sendTransaction(options);
     }
 
-    var result = {};
-    addFunctionRelatedPropertiesToContract(result);
-    addFunctionsToContract(result, abi, address);
-    addEventRelatedPropertiesToContract(result, abi, address);
-    addEventsToContract(result, abi, address);
+    addFunctionsToContract(this, abi);
+    addEventsToContract(this, abi, this.address);
+};
+
+Contract.prototype.call = function (options) {
+    this._isTransaction = false;
+    this._options = options;
+    return this;
+};
```

Comment:
Is there a way that we can structure this so that we call:
`myCotract.myMethod().sendTransaction()` it looks more natural to me and looks more logical structured.